catroid/src/main/java/org/catrobat/catroid/content/bricks/BroadcastMessageBrick.java

```diff
@@ -83,15 +83,15 @@ public View getView(Context context) {
 
 	@Override
 	public void onNewOptionSelected() {
-		AppCompatActivity activity = UiUtils.getActivityFromView(view);
+		final SpriteActivity activity = (SpriteActivity) UiUtils.getActivityFromView(view);
```

Comment:
please change to 
```
AppCompatActivity activity = UiUtils.getActivityFromView(view);
if (!(activity instanceof SpriteActivity)) {
    retrun;
}
```
because otherwise this is going to crash with a `ClassCastException` if the activity is not a `SpriteActivity`. and `instanceof` checks if it is `null` as well.

Reply:
since you now rely on the parent activity being the `SpriteActivity` we could probably find another way to notify the other bricks to update their spinners. namely: we could just move this dialog to the `SpriteActivity` as a whole and then register this brick as listener, then we could call `refreshScriptFragment()` before or immediately after we call back to this brick from the `SpriteActivity`.

then you could get rid of `void notifyDataSetChanged(AppCompatActivity activity)` in the `BrickBaseType`, which I would absolutely prefer. plus, then it would be the same as for adding new looks or sprites etc.

Reply:
> please change to
> 
> ```
> AppCompatActivity activity = UiUtils.getActivityFromView(view);
> if (!(activity instanceof SpriteActivity)) {
>     retrun;
> }
> ```
> because otherwise this is going to crash with a `ClassCastException` if the activity is not a `SpriteActivity`. and `instanceof` checks if it is `null` as well.

done.

Reply:
> since you now rely on the parent activity being the `SpriteActivity` we could probably find another way to notify the other bricks to update their spinners. namely: we could just move this dialog to the `SpriteActivity` as a whole and then register this brick as listener, then we could call `refreshScriptFragment()` before or immediately after we call back to this brick from the `SpriteActivity`.
> 
> then you could get rid of `void notifyDataSetChanged(AppCompatActivity activity)` in the `BrickBaseType`, which I would absolutely prefer. plus, then it would be the same as for adding new looks or sprites etc.

since `void notifyDataSetChanged(AppCompatActivity activity)` has another 2 usages in `(SceneStartBrick.java, SceneTransitionBrick.java)`, I disagree